agent/engine/docker_task_engine_linux.go

```diff
@@ -39,3 +44,78 @@ func (engine *DockerTaskEngine) updateTaskENIDependencies(task *apitask.Task) {
 func (engine *DockerTaskEngine) invokePluginsForContainer(task *apitask.Task, container *apicontainer.Container) error {
 	return nil
 }
+
+func (engine *DockerTaskEngine) watchAppNetImage(ctx context.Context, filepath string) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		seelog.Errorf("error adding %s to fsnotify watcher, error: %v", filepath, err)
+	}
+	err = watcher.Add(filepath)
+	if err != nil {
+		seelog.Errorf("failed to initialize fsnotify NewWatcher, error: %v", err)
+	}
+	defer watcher.Close()
+	engine.watch(ctx, watcher)
+}
+
+func (engine *DockerTaskEngine) watch(ctx context.Context, watcher *fsnotify.Watcher) {
+	// Start listening for events.
+	for {
+		select {
+		case event, ok := <-watcher.Events:
+			if !ok {
+				seelog.Warn("fsnotify watcher event error")
```

Comment:
`!ok` indicates that the channel has been closed, not necessarily an error event.